modules/Microsoft.EventGrid/domainTopics/deploy.bicep

```diff
@@ -0,0 +1,57 @@
+@description('Required. The name of the Event Grid Domain Topic.')
+param name string
+
+@description('Required. Name of the Event Grid Domain.')
+param domainName string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments array = []
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name, location)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource domain 'Microsoft.EventGrid/domains@2022-06-15' existing = {
```

Comment:
The intension of the test is to validate the module. Here we're deploying a domain and its child resource topic directly through bicep resources. The test should use the module instead. 

I'd suggest to start only with the parent resource `domains` and complete the PR for that. Then as a second PR we can extend the module and the test with the `topics` child resource.

Reply:
We have already domains module. Customer is looking for module to deploy topics under domain. This module will help to customer to create topics under domains.

Reply:
Same as previous comment. `Topics` should be added as a child module of the already existing `domains`

Reply:
I have removed the topics independent module and create child module for the same topics.

can you please review